modules/aruba-sdwan/variables.tf

```diff
@@ -0,0 +1,216 @@
+  description = "Device specific vendor configurations."
+  type = object({
+    accountKey   = string
+    accountName  = string
+    applianceTag = string
+    hostname     = string
+  })
+
+  validation {
+    condition     = try(length(var.vendor_configuration.accountKey) > 0, false)
+    error_message = "Account Key has to be a non empty string."
+  }
+
+  validation {
+    condition     = try(length(var.vendor_configuration.accountName) > 0, false)
+    error_message = "Account Name has to be a non empty string."
+  }
+
+  validation {
+    condition     = try(length(var.vendor_configuration.applianceTag) > 0, false)
+    error_message = "Appliance Tag has to be a non empty string."
+  }
+
+  validation {
+    condition     = try(length(var.vendor_configuration.hostname) > 0, false)
+    error_message = "hostname has to be a non empty string."
+  }
+}
+
+variable "package_code" {
+  description = "Device software package"
+  type        = string
+  validation {
+    condition     = can(regex("^(EC-V)$", var.package_code))
+    error_message = "One of following software packages are supported: EC-V."
+  }
+}
+
+variable "version_number" {
+  description = "version number"
+  type        = string
+  default     = ""
+}
+
+variable "name" {
+  description = "Device name"
+  type        = string
+  validation {
+    condition     = length(var.name) >= 3 && length(var.name) <= 50
+    error_message = "Device name should consist of 3 to 50 characters."
+  }
+}
+
+variable "term_length" {
+  description = "Term length in months"
+  type        = number
+  validation {
+    condition     = can(regex("^(1|12|24|36)$", var.term_length))
+    error_message = "One of following term lengths are available: 1, 12, 24, 36 months."
+  }
+}
+
+variable "notifications" {
+  description = "List of email addresses that will receive device status notifications"
+  type        = list(string)
+  validation {
+    condition     = length(var.notifications) > 0
+    error_message = "Notification list cannot be empty."
+  }
+}
+
+variable "acl_template_id" {
+  description = "Identifier of an ACL template that will be applied on a device"
+  type        = string
+  validation {
+    condition     = try(length(var.acl_template_id) > 0, false)
+    error_message = "Acl template is required."
+  }
+}
+
+variable "additional_bandwidth" {
+  description = "Additional internet bandwidth for a device"
+  type        = number
+  default     = 0
+  validation {
+    condition     = var.additional_bandwidth == 0 || (var.additional_bandwidth >= 25 && var.additional_bandwidth <= 5001)
+    error_message = "Additional internet bandwidth should be between 25 and 5001 Mbps."
+  }
+}
+
+variable "interface_count" {
+  description = "Number of network interfaces on a device. If not specified, default number for a given device type will be used."
+  type        = number
+  default     = 10
+  validation {
+    condition     = can(regex("^(10)$", var.interface_count))
```

Comment:
For Aruba device, both 10 and 32 interfaces are allowed. 
Please refer to this page for more information.
https://docs.equinix.com/network-edge/vendors-devices/aruba/ne-aruba-specs